### DISCUSSION TITLE
Any way to stop rerunning completely?

### DISCUSSION THREAD

**Discussion by Apon77** [cirruslabs/cirrus-ci-docs#941] [Ideas]
Dear sir,
Is there any way to stop rerunning a build completely (with the help of .cirrus.yml file). 
We really need this feature for our persistence worker server. Please help. 
Thank you.

**Comment by RDIL** [comment]
Automatic reruns or manual (user or API triggered) reruns?

**Reply by Apon77** [reply]
All kind of rerun we want to stop..
Because if someone do force push, it starts reruning, and we got confused about current build.. It hampers our other users queue which makes our persistent worker work slower than expected.. Please do something.. We need this very much..
Thank you.

**Reply by Apon77** [reply]
and github even shows a rerun button, we want to remove rerun option completely from our repository. please help us regarding this.

**Reply by Apon77** [reply]
> Automatic reruns or manual (user or API triggered) reruns?

Dear Sir, Please do something for the rerun thing. Our resources is getting misused by rerun by pull requests. @edigaryev

**Comment by edigaryev** [comment]
Have you considered using https://cirrus-ci.org/guide/writing-tasks/#required-pr-labels to solve this?

Note that if someone is still allowed to trigger builds in your repository despite having no "Re-run" button, they might still trigger builds by doing the force pushes you've mentioned above, since to the GitHub and CI view this is a new code (it's SHA is different).

**Reply by Apon77** [reply]
> Have you considered using https://cirrus-ci.org/guide/writing-tasks/#required-pr-labels to solve this?
> 
> Note that if someone is still allowed to trigger builds in your repository despite having no "Re-run" button, they might still trigger builds by doing the force pushes you've mentioned above, since to the GitHub and CI view this is a new code (it's SHA is different).

That link probably wont work, because previous builds uses .cirrus.yml file from previous .cirrus.yml as far I know. 

or, is there any way to running build and cancel a bulid which is ran by reran? then i can setup another script to check this by CI and cancel the required build by terminal.
Probably this may be done by API or something. But I really could not find how to do this. Can you please help me to do this (from terminal, so that it can be scripted)?? 
Thank you.

**Reply by edigaryev** [reply]
>That link probably wont work, because previous builds uses .cirrus.yml file from previous .cirrus.yml as far I know.

There's also an approval setting in the repository's settings:

<img width="483" alt="Screen Shot 2021-11-16 at 23 18 38" src="https://user-images.githubusercontent.com/85709/142059336-1a0712d9-d281-4632-9985-f6e7695ce43a.png">

Does that work for you?
